manager/scheduler/filter.go

```diff
@@ -217,3 +217,48 @@ func (f *ConstraintFilter) Explain(nodes int) string {
 	}
 	return fmt.Sprintf("scheduling constraints not satisfied on %d nodes", nodes)
 }
+
+// PlatformFilter selects only nodes that run the required platform.
+type PlatformFilter struct {
+	supportedPlatforms []*api.Platform
+}
+
+// SetTask returns true when the filter is enabled for a given task.
+func (f *PlatformFilter) SetTask(t *api.Task) bool {
+	c := t.Spec.GetContainer()
+	if c != nil {
+		if len(c.SupportedPlatforms) > 0 {
+			// copy the platform information
+			for _, p := range c.SupportedPlatforms {
+				f.supportedPlatforms = append(f.supportedPlatforms, p)
+			}
```

Comment:
@aaronlehmann updated to remove the for loop. With using `f.supportedPlatforms = c.SupportedPlatforms`, we don't need to reset it anymore, right?